apps/els_lsp/src/els_dt_document.erl

```diff
@@ -145,18 +145,21 @@ new(Uri, Text) ->
 
 -spec new(uri(), binary(), atom(), kind()) -> item().
 new(Uri, Text, Id, Kind) ->
-  {ok, POIs} = els_parser:parse(Text),
-  MD5        = erlang:md5(Text),
+  MD5 = erlang:md5(Text),
   #{ uri  => Uri
    , id   => Id
    , kind => Kind
    , text => Text
```

Comment:
is it necessary to store the Text if the POIs are not (yet) stored? if Text is still stored it could be used to grep in the in-memory binary. If it's not stored it can spare some memory.

Reply:
Switched to an in-memory search (that can definitely be optimized) instead of the grep.